# 01_Inspección inicial

In [ ]:
# Importación de librerías
import pandas as pd
import numpy as np

In [ ]:
# Carga del dataset

url = "/content/streaming_users_dirty.json"
df = pd.read_json(url)
df.head()

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1


In [ ]:
df.shape

(8160, 8)

Mediante esta exploración inicial podemos determinar que el dataset tiene 8160 filas y 8 columnas.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   object 
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   object 
 5   favorite_genre            7920 non-null   object 
 6   last_login_date           7840 non-null   object 
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), object(4)
memory usage: 510.1+ KB


En el dataset se pueden observar variables numéricas y categóricas: tres columnas con números enteros; una columna con números decimales y cuatro columnas de texto o genéricas. También se puede observar que la memoria utilizada es de 510.1+ KB. Asimismo, podemos observar que hay valores faltantes en las siguientes columnas: monthly_watch_time_mins; favorite_genre; y last_login_date.

In [ ]:
df.isnull().sum()

,0
user_id,0
age,0
subscription_plan,0
monthly_watch_time_mins,193
country,0
favorite_genre,240
last_login_date,320
customer_support_tickets,0


La cantidad de datos nulos en las columnas antes mencionadas son:
-monthly_watch_time_mins: 193 datos nulos
-favorite_genre: 240 datos nulos
-last_login_date: 320 datos nulos


In [ ]:
df.describe()

,age,monthly_watch_time_mins,last_login_date,customer_support_tickets
count,8160.000000,7967.000000,7776,8160.000000
mean,34.096814,1107.346153,2022-02-05 04:58:20,1.800980
min,-5.000000,-120.000000,2018-01-01 00:00:00,-1.000000
25%,25.000000,489.200000,2020-01-23 12:00:00,0.000000
50%,33.000000,757.400000,2022-02-15 12:00:00,1.000000
75%,42.000000,1045.700000,2024-02-20 06:00:00,1.000000
max,150.000000,99999.000000,2029-01-01 00:00:00,150.000000
std,14.511304,5310.442604,NaN,11.334969


La función df.describe() muestra el resumen estadístico de las columnas numéricas, donde se pueden identificar anomalías, errores lógicos y valores atípicos (outliers) en los datos que deben ser corregidos antes de realizar cualquier análisis. Por ejemplo, en la columna 'age' el valor mínimo es -5 y el valor máximo es 150, lo que es biologicamente imposible y generaría fallas en el análisis. En la columna 'montthly_watch_time' (tiempo de reproducción mensual en minutos) se registra un valor mínimo negativo de -120 y un máximo de 99999 lo que supera a los minutos máximos de un mes. En la columna last_login_date el valor máximo es 2029, lo que no se corresponde al tiempo real ya que estamos actualmente en el año 2026. En la columna 'customer_support_tickets' el valor mínimo es de -1 y el máximo de 150, lo cual representa resultados anómalos ya que, por una parte, no se puede abrir -1 tickets y, por otra parte, 75% de los usuarios tiene un máximo de 1 ticket, representando el 150 un outlier (valor atípico) extremo.

In [ ]:
df.duplicated().sum()

np.int64(126)

Con esta función, podemos identificar la cantidad de registros duplicados o clonados que presenta el detaset.

In [ ]:
for columna in df.columns:
  print (columna)
  print (df[columna].unique())
  print ("-"*40)

user_id
[10000 10001 10002 ... 17997 17998 17999]
----------------------------------------
age
[ 39  37  28  43  51  20  31  36  23  46  15  45  29  24  35  32  30  17
  33  18  40  42  22  44  41  59  26  25  27  53   0  55  34  47  16  21
  13  38  56  19  48  14  57  52  64  60  58 130  54  50  49  61  66  73
 150  -5  68   4  69  62  72  63  67  65  76  80  74  71  70]
----------------------------------------
subscription_plan
['Estándar' 'Básico' 'Premium' 'Std' 'estandar' 'basico' 'básico'
 'Premium ' 'premium' 'Premiun' 'BASICO' 'STANDARD' 'Basic' 'Estándar '
 'PREMIUM']
----------------------------------------
monthly_watch_time_mins
[ 805.8 1173.4  401.  ...  410.4  501.4 1574.8]
----------------------------------------
country
['Brasil' 'Colombia' 'Uruguay' 'Perú' 'Chile' 'Argentina' 'México'
 'Brazil' 'brasil' 'méxico' 'chile' 'uruguay' 'MEX' 'ARG' 'colombia' 'COL'
 'Mexico' 'URY' 'Chile ' 'Peru' 'perú' 'argentina' 'PER' 'Argentina '
 'BRA' 'CHL']
---------------------------

La exploración de valores únicos nos muestra que, además de los valores atípicos (outliers), el dataset presenta una falta de estandarización en las columnas categóricas. Por ejemplo, en la columna 'suscription_plan' la misma variable fue introducida de diferente manera incluyendo errores como mezcla de mayusculas y minusculas, abreviaciones, errores de escritura y espacios vacios al final de las palabras. En la columna 'country' también se observa la duplicación de categorías por la falta de un formato único. Por último, en la columna 'favorite_genre' se observan los errores anteriores como la mezcla de mayuscula y minuscula; una mezcla entre idiomas y la presencia de valores nulos con el texto 'None'. Asimismo, se confirman los valores atípicos obtenidos en el 'df.describe' y se identifican otros. Por ejemplo en la columna 'age' además de los outliers -5 y 150, aparece un valor de 130, lo que también es biologicamente imposible. Debido a esto, es necesario aplicar técnicas de estandarización para obtener un dataset listo para análisis.